src/appState.ts

```diff
@@ -62,30 +62,86 @@ export const getDefaultAppState = (): AppState => {
   };
 };
 
-export const clearAppStateForLocalStorage = (appState: AppState) => {
-  const {
-    draggingElement,
-    resizingElement,
-    multiElement,
-    editingElement,
-    selectionElement,
-    isResizing,
-    isRotating,
-    collaborators,
-    isCollaborating,
-    isLoading,
-    errorMessage,
-    showShortcutsDialog,
-    editingLinearElement,
-    isLibraryOpen,
-    ...exportedState
-  } = appState;
-  return exportedState;
+const APP_STATE_STORAGE_CONF = (<
+  Values extends { browser: boolean; export: boolean },
+  T extends Record<keyof AppState, Values>
+>(
+  config: { [K in keyof T]: K extends keyof AppState ? T[K] : never },
+) => config)({
+  isLoading: { browser: false, export: false },
+  errorMessage: { browser: false, export: false },
+  draggingElement: { browser: false, export: false },
+  resizingElement: { browser: false, export: false },
+  multiElement: { browser: false, export: false },
+  editingElement: { browser: false, export: false },
+  editingLinearElement: { browser: false, export: false },
+  isCollaborating: { browser: false, export: false },
+  isResizing: { browser: false, export: false },
+  isRotating: { browser: false, export: false },
+  selectionElement: { browser: false, export: false },
+  collaborators: { browser: false, export: false },
+  showShortcutsDialog: { browser: false, export: false },
+  width: { browser: false, export: false },
+  height: { browser: false, export: false },
+  elementType: { browser: true, export: false },
+  elementLocked: { browser: true, export: false },
+  exportBackground: { browser: true, export: false },
+  shouldAddWatermark: { browser: true, export: false },
+  currentItemStrokeColor: { browser: true, export: false },
+  currentItemBackgroundColor: { browser: true, export: false },
+  currentItemFillStyle: { browser: true, export: false },
+  currentItemStrokeWidth: { browser: true, export: false },
+  currentItemStrokeStyle: { browser: true, export: false },
+  currentItemRoughness: { browser: true, export: false },
+  currentItemOpacity: { browser: true, export: false },
+  currentItemFontSize: { browser: true, export: false },
+  currentItemFontFamily: { browser: true, export: false },
+  currentItemTextAlign: { browser: true, export: false },
+  viewBackgroundColor: { browser: true, export: true },
+  scrollX: { browser: true, export: false },
+  scrollY: { browser: true, export: false },
+  cursorX: { browser: true, export: false },
+  cursorY: { browser: true, export: false },
+  cursorButton: { browser: true, export: false },
+  scrolledOutside: { browser: true, export: false },
+  name: { browser: true, export: false },
+  username: { browser: true, export: false },
+  zoom: { browser: true, export: false },
+  openMenu: { browser: true, export: false },
+  lastPointerDownWith: { browser: true, export: false },
+  selectedElementIds: { browser: true, export: false },
+  previousSelectedElementIds: { browser: true, export: false },
```

Comment:
can we remove this and `selectedElementIds` as well from the browser ? I am not sure if this is helping the user

Reply:
Now that I think about it, it may be useful when we hard-refresh during SW update. Either way, I'd change the defaults in a separate PR.